spacy_llm/tasks/ner.py

```diff
@@ -11,69 +12,31 @@
-from .span import SpanExample, SpanTask
+from .span import SpanExample, SpanReason, SpanTask
 from .templates import read_template
 
 _DEFAULT_NER_TEMPLATE_V1 = read_template("ner")
 _DEFAULT_NER_TEMPLATE_V2 = read_template("ner.v2")
+_DEFAULT_NER_TEMPLATE_V3 = read_template("ner.v3")
 
 
-@registry.llm_tasks("spacy.NER.v1")
-def make_ner_task(
-    labels: str = "",
-    examples: Optional[Callable[[], Iterable[Any]]] = None,
-    normalizer: Optional[Callable[[str], str]] = None,
-    alignment_mode: Literal["strict", "contract", "expand"] = "contract",
-    case_sensitive_matching: bool = False,
-    single_match: bool = False,
-):
-    """NER.v1 task factory.
-
-    labels (str): Comma-separated list of labels to pass to the template.
-        Leave empty to populate it at initialization time (only if examples are provided).
-    template (str): Prompt template passed to the model.
-    label_definitions (Optional[Dict[str, str]]): Map of label -> description
-        of the label to help the language model output the entities wanted.
-        It is usually easier to provide these definitions rather than
-        full examples, although both can be provided.
-    examples (Optional[Callable[[], Iterable[Any]]]): Optional callable that
-        reads a file containing task examples for few-shot learning. If None is
-        passed, then zero-shot learning will be used.
-    normalizer (Optional[Callable[[str], str]]): optional normalizer function.
-    alignment_mode (str): "strict", "contract" or "expand".
-    case_sensitive: Whether to search without case sensitivity.
-    single_match (bool): If False, allow one substring to match multiple times in
-        the text. If True, returns the first hit.
-    """
-    labels_list = split_labels(labels)
-
-    span_examples = (
-        [SpanExample(**eg) for eg in examples()] if callable(examples) else examples
-    )
-    return NERTask(
-        labels=labels_list,
-        template=_DEFAULT_NER_TEMPLATE_V1,
-        prompt_examples=span_examples,
-        normalizer=normalizer,
-        alignment_mode=alignment_mode,
-        case_sensitive_matching=case_sensitive_matching,
-        single_match=single_match,
-    )
-
-
-@registry.llm_tasks("spacy.NER.v2")
-def make_ner_task_v2(
+@registry.llm_tasks("spacy.NER.v3")
+def make_ner_task_v3(
+    examples: ExamplesConfigType,
     labels: Union[List[str], str] = [],
-    template: str = _DEFAULT_NER_TEMPLATE_V2,
+    template: str = _DEFAULT_NER_TEMPLATE_V3,
+    description: Optional[str] = None,
     label_definitions: Optional[Dict[str, str]] = None,
-    examples: ExamplesConfigType = None,
     normalizer: Optional[Callable[[str], str]] = None,
     alignment_mode: Literal["strict", "contract", "expand"] = "contract",
     case_sensitive_matching: bool = False,
     single_match: bool = False,
 ):
-    """NER.v2 task factory.
+    """NER.v3 task factory.
 
+    examples (Union[Callable[[], Iterable[OTS]]]): Optional callable that
```

Comment:
Where's `OTS` coming from?

Reply:
Not sure, this wasn't me - @kabirkhan?

Reply:
Wasn't me either I don't think haha. This should be Just be `ExamplesConfigType`